paasta_tools/mesos_tools.py

```diff
@@ -165,7 +170,12 @@ def find_mesos_leader(cluster):
         raise ValueError("Mesos master is required to find leader")
 
     url = f"http://{master}:{MESOS_MASTER_PORT}/redirect"
-    response = requests.get(url)
+    try:
+        # Timeouts here are for connect, read
+        response = requests.get(url, timeout=(5, 30))
+    except Exception as e:
+        print(f"Error finding the mesos leader: {e}", file=sys.stderr)
+        raise MesosLeaderUnavailable(e)
```

Comment:
Embedding the original exception inside `MesosLeaderUnavailable` will obscure the original exception in a print-out. I think it should be `raise MesosLeaderUnavailable() from e` instead.

Reply:
I don't really understand. My motivation here is that the original error (timeout, failed to connect, whatever it is) still gets printed to stderr in some way.

If I do `raise MLU from e` here, would you give a shipit if I printed the `e` in paasta spark run?

Reply:
My motivation for chaining here is that I'd like to see the original exception class itself printed as well as well as the traceback to see where it happened. Often times, logging only the message is not quite enough for debugging, though this case it's less important.

Reply:
Hm, given the output, I'm fine without the chaining. As long as the original exception is visible, even without a traceback, I'm happy with it.